File: source/code/vbnet/authentication/create_directory_github.vb
Imports Stormpath.SDK.Provider

Dim githubDirectory = client.Instantiate(Of IDirectory)()
githubDirectory.SetName("My Github Directory")

Await client.CreateDirectoryAsync(
    githubDirectory,
    Function(options) options.ForProvider(client _
                                          .Providers() _
                                          .Github() _
                                          .Builder() _
                                          .SetClientId("YOUR_GITHUB_CLIENT_ID") _
                                          .SetClientSecret("YOUR_GITHUB_CLIENT_SECRET") _
                                          .Build()))

File: source/code/nodejs/authentication/create_directory_github.js
var gitHubDirectory = {
  name : 'my-github-directory',
  description : 'A GitHub directory',
  provider: {
    providerId: 'github',
    clientId: 'YOUR_GITHUB_CLIENT_ID',
    clientSecret: 'YOUR_GITHUB_CLIENT_SECRET'
  }
};

client.createDirectory(gitHubDirectory, function (err, directory) {
  if (err) {
    return console.error(err);
  }

  console.log('GitHub directory created!');
});